pandas/core/reshape/merge.py

```diff
@@ -1015,6 +1015,43 @@ def _get_join_info(
         left_ax = self.left.axes[self.axis]
         right_ax = self.right.axes[self.axis]
 
+        if self.how in ["anti_left", "anti_right", "anti_full"]:
+            how_dict = {
+                "anti_left": "left",
+                "anti_right": "right",
+                "anti_full": "outer",
+            }
+            self.how = how_dict[self.how]
```

Comment:
In the previous PR, I had changed `_MergeOperation` so that anti's are converted into left/right/outer and relevant changes are made in the initial dataframes. 
But Jeff had suggested https://github.com/pandas-dev/pandas/pull/43056#pullrequestreview-763630935 that we should keep the validation and the merge operation separate (as it is currently for the other joins).

The final result is same as before. I think I tweaked a couple of test results, but I guess those were bugs that were present earlier.

Reply:
I think I missed, so the side effect should not be in the validation.. this is in the operation